packages/ai-chat-ui/src/browser/chat-response-renderer/markdown-part-renderer.tsx

```diff
@@ -66,16 +66,17 @@ const MarkdownRender = ({ response }: { response: MarkdownChatResponseContent |
  * This leads to unexpected behavior when rendering markdown with html tags.
  *
  * @param markdown the string to render as markdown
+ * @param skipSurroundingParagraph whether plain text should be surrounded by a paragraph (default: false)
  * @returns the ref to use in an element to render the markdown
  */
-export const useMarkdownRendering = (markdown: string | MarkdownString) => {
+export const useMarkdownRendering = (markdown: string | MarkdownString, skipSurroundingParagraph: boolean = false) => {
     // eslint-disable-next-line no-null/no-null
     const ref = useRef<HTMLDivElement | null>(null);
     const markdownString = typeof markdown === 'string' ? markdown : markdown.value;
     useEffect(() => {
         const markdownIt = markdownit();
         const host = document.createElement('div');
-        const html = markdownIt.render(markdownString);
+        const html = skipSurroundingParagraph ? markdownIt.render(markdownString).replace(/^<p>|<\/p>|<p><\/p>$/g, '') : markdownIt.render(markdownString);
```

Comment:
Why is the `skipSurroundingParagraph` needed?

Reply:
Because for plain text the markdown renderer always surrounds the text with a `<p>`. For the request rendering, where we programmatically join the parsed parts, we [surround this with a `<p>`](https://github.com/eclipse-theia/theia/pull/14498/files#diff-406d81bf9c76fbc86341292ee64753d2170002ce325fe71f1bfe9d3242a55462R445) so that all (parsed parts and normal parts) are in one paragraph.

There is a `renderInline` method I used before, but this then breaks support for enumerations and things like that.